modules/frontend/combiner/metrics_query_range.go

```diff
@@ -0,0 +1,85 @@
+package combiner
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/pkg/traceql"
+)
+
+var _ GRPCCombiner[*tempopb.QueryRangeResponse] = (*genericCombiner[*tempopb.QueryRangeResponse])(nil)
+
+// NewQueryRange returns a query range combiner.
+func NewQueryRange() Combiner {
+	combiner := traceql.QueryRangeCombiner{}
+
+	return &genericCombiner[*tempopb.QueryRangeResponse]{
+		httpStatusCode: 200,
+		new:            func() *tempopb.QueryRangeResponse { return &tempopb.QueryRangeResponse{} },
+		current:        &tempopb.QueryRangeResponse{Metrics: &tempopb.SearchMetrics{}},
+		combine: func(partial *tempopb.QueryRangeResponse, _ *tempopb.QueryRangeResponse, resp PipelineResponse) error {
+			if partial.Metrics != nil {
+				// this is a coordination between the sharder and combiner. the sharder returns one response with summary metrics
+				// only. the combiner correctly takes and accumulates that job. however, if the response has no jobs this is
+				// an indicator this is a "real" response so we set CompletedJobs to 1 to increment in the combiner.
+				if partial.Metrics.TotalJobs == 0 {
+					partial.Metrics.CompletedJobs = 1
+				}
+			}
+
+			samplingRate := resp.AdditionalData()
+			if samplingRate != nil {
+				fRate := samplingRate.(float64)
+
+				if fRate != 0.0 {
```

Comment:
nit:  can skip this when rate == 1.0.  Maybe switch to `fRate < 1.0` ?

Reply:
the 0.0 check is just a sanity check to prevent panics. `fRate < 1.0 && fRate > 0.0`. would there ever be a time we deflated metrics with the rate?

Reply:
No, the sampling rate will always be between 0 and 1 due to the check in the sharder: https://github.com/grafana/tempo/pull/3584/files#diff-85974350e62b77bee6b5d88d537e3946839ccb1bba5f6dc89450a1c5489b5d4fR364

Reply:
good catch. updated to the suggested condition